maven-core/src/main/java/org/apache/maven/plugin/DefaultExtensionRealmCache.java

```diff
@@ -41,41 +41,46 @@
 public class DefaultExtensionRealmCache
     implements ExtensionRealmCache, Disposable
 {
+
+    private static final ConcurrentHashMap<String, FileInfo> FILE_INFO_CACHE = new ConcurrentHashMap<>( 512 );
+
     /**
      * CacheKey
      */
     protected static class CacheKey
         implements Key
     {
 
-        private final List<File> files;
-
-        private final List<Long> timestamps;
-
-        private final List<Long> sizes;
-
-        private final List<String> ids;
+        private final List<FileInfo> files;
 
         private final int hashCode;
 
         public CacheKey( List<Artifact> extensionArtifacts )
         {
             this.files = new ArrayList<>( extensionArtifacts.size() );
-            this.timestamps = new ArrayList<>( extensionArtifacts.size() );
-            this.sizes = new ArrayList<>( extensionArtifacts.size() );
-            this.ids = new ArrayList<>( extensionArtifacts.size() );
 
             for ( Artifact artifact : extensionArtifacts )
             {
-                File file = artifact.getFile();
-                files.add( file );
-                timestamps.add( ( file != null ) ? Long.valueOf( file.lastModified() ) : Long.valueOf( 0 ) );
-                sizes.add( ( file != null ) ? Long.valueOf( file.length() ) : Long.valueOf( 0 ) );
-                ids.add( artifact.getVersion() );
+                String artifactKey = getArtifactKey( artifact );
+                FileInfo fileInfo = FILE_INFO_CACHE.get( artifactKey );
```

Comment:
Hi. By the time i i implemented this maven was on java 7. will update to v.8 api shortly

Reply:
Hi reworked a bit - as it doesn't make much sense to access file system at all, i removed that part. to restore previous behavior in case of issues new command line flag introduced